pkg/build/util/util.go

```diff
@@ -307,3 +307,41 @@ func SafeForLoggingS2IConfig(config *s2iapi.Config) *s2iapi.Config {
 	newConfig.ScriptsURL, _ = s2iutil.SafeForLoggingURL(newConfig.ScriptsURL)
 	return &newConfig
 }
+
+// GetBuildEnv gets the build strategy environment
+func GetBuildEnv(build *buildapi.Build) *[]kapi.EnvVar {
+	switch {
+	case build.Spec.Strategy.SourceStrategy != nil:
+		return &build.Spec.Strategy.SourceStrategy.Env
+	case build.Spec.Strategy.DockerStrategy != nil:
+		return &build.Spec.Strategy.DockerStrategy.Env
+	case build.Spec.Strategy.CustomStrategy != nil:
+		return &build.Spec.Strategy.CustomStrategy.Env
+	case build.Spec.Strategy.JenkinsPipelineStrategy != nil:
+		return &build.Spec.Strategy.JenkinsPipelineStrategy.Env
+	default:
+		return nil
+	}
+}
+
+// UpdateBuildEnv updates the strategy environment
+// This will replace the existing variable definitions with provided env
+func UpdateBuildEnv(build *buildapi.Build, env []kapi.EnvVar) {
```

Comment:
Apart from being O(m*n), this function is pretty dodgy if env has any ValueFroms set.  I think either we need to disallow ValueFrom being set in a BuildRequest (I think this is intended, but maybe not actually done today) or this function needs enhancement, or a bit of both.  @bparees?